.buildkite/steps/publish-docker-image.sh

```diff
@@ -40,6 +40,9 @@ release_image() {
   dry_run skopeo copy --multi-arch all "docker://${source_image}" "docker://docker.io/${target_image}:${tag}"
   echo "--- :github: Copying ${target_image}:${tag} to GHCR"
   dry_run skopeo copy --multi-arch all "docker://${source_image}" "docker://ghcr.io/${target_image}:${tag}"
+  echo "--- :buildkite: Copying ${target_image}:${tag} to Buildkite Packages"
+  dry_run skopeo copy --multi-arch all "docker://${source_image}" "docker://packages.buildkite.com/buildkite/agent-docker/agent/${target_image}:${tag}" \
+    || true # Soft fail if we can't upload to Packages
```

Comment:
```suggestion
  dry_run skopeo copy --multi-arch all "docker://${source_image}" "docker://packages.buildkite.com/buildkite/agent-docker/agent/${target_image}:${tag}"
```

i think that soft-failing the step when pushing to packages is a good call, but my worry is that will just silently fail, and we don't notice. i reckon we should either: 
- make it so that (as above) this hard-fails the build - and we can go and whinge to packages that docker registries are broken - or, 
- have a separate step that pushes docker images to packages, and allow that step to soft fail

with both of the above strategies, there's strong signals that pushing docker images to BK packages has failed, and we can come and talk to y'all about it

Reply:
I'd like to build a little confidence it in before switching to hard fail. How about we leave it as soft-fail for a week and get a few builds pushing through successfully and then switch to hard fail?

Reply:
I completely agree with @moskyb that this could just start failing one day and nobody would notice. I doubt we need to make it soft-fail either. If it doesn't work, it's not difficult to revert and try again.

Reply:
I've changed this so it will hard fail.